lib/modules/manager/osgifeature/extract.ts

```diff
@@ -0,0 +1,84 @@
+import * as json5 from 'json5';
+import { logger } from '../../../logger';
+import { MavenDatasource } from '../../datasource/maven';
+import type { ExtractConfig, PackageDependency, PackageFile } from '../types';
+
+export function extractPackageFile(
+  content: string,
+  fileName: string,
+  config?: ExtractConfig
+): PackageFile | null {
+  // References:
+  // - OSGi compendium release 8 ( https://docs.osgi.org/specification/osgi.cmpn/8.0.0/service.feature.html )
+  // - The Sling implementation of the feature model ( https://sling.apache.org/documentation/development/feature-model.html )
+  logger.debug({ fileName }, 'osgifeature.extractPackageFile');
+
+  const deps: PackageDependency[] = [];
+  try {
+    // Compendium R8 159.3: JS comments are supported
+    const featureModel = json5.parse(content);
```

Comment:
@secustor - by a typed approach do you mean using an interface/class to represent the complete feature model definition? I id not do that for two reasons:

1. The model is extensible, with dynamic property names, which makes it hard to do proper typing. As you can see in the PR, we check for certain section names
2. The model is richer than just describing dependencies, it can also hold configurations, text section ( e.g. DDL for an SQL database ) and I think representing all of that is not needed

Reply:
You would have to add only the attributes you actually need for Renovate. 

e.g. 
```ts
export interface MyModel {
  bundles?: Bundle[];
  'execution-environment:JSON|false'?: {
    'framework?': Bundle
  }
  [x: string]: any;
}

export type Bundle = string | BundleObject;

export interface BundleObject {
  id: string;
}
``` 

But I'm fine to approve it within the current scope if the whole json is typed, but the processing should be typed anyway.

Reply:
There is no way for fully dynamic ones, but templating is possible. 

```ts
type keys = 'content-packages' | 'other-packages' 
export type Artifacts = `${keys}:ARTIFACTS|${boolean}`

// the same as 
export type Templated = `content-packages:ARTIFACTS|true` |
                        `content-packages:ARTIFACTS|false` |
                        `other-packages:ARTIFACTS|true` |
                        `other-packages:ARTIFACTS|false` |
```

You will have to filter entries, based on your conditions.